schematic/store/synapse.py

```diff
@@ -1401,14 +1402,16 @@ def format_row_annotations(
             else:
                 if isinstance(anno_v, float) and np.isnan(anno_v):
                     annos[anno_k] = ""
-                elif (
-                    isinstance(anno_v, str)
-                    and re.fullmatch(csv_list_regex, anno_v)
-                    and rule_in_rule_list(
-                        "list", dmge.get_node_validation_rules(anno_k)
-                    )
-                ):
-                    annos[anno_k] = anno_v.split(",")
+                elif isinstance(anno_v, str) and re.fullmatch(csv_list_regex, anno_v):
```

Comment:
this is too many layers of elif and if else underneath. I would prefer this part to be refactored to be more maintainable.